core/src/main/scala/cats/syntax/parallel.scala

```diff
@@ -3,6 +3,9 @@ package cats.syntax
 import cats.{FlatMap, Foldable, Monad, Parallel, Traverse}
 
 trait ParallelSyntax extends TupleParallelSyntax {
+  implicit final def catsSyntaxParallelApply[F[_], A, B](fa: F[A => B]): ParallelApplyOps[F, A, B] =
```

Comment:
Unfortunately, you can't add a method to a trait without breaking binary compatibility in Scala 2.11. 
You need to add a new trait to add this method and have relevant objects extending it. And then you can have `AllSyntaxBinCompat4` (a new trait introduced post 1.5.0 release) extend this.